### PR TITLE
#10 Publish events about new messages ready to be processed after the transaction commit

### PR DESCRIPTION
**Pull request by mcybulsk**

### CODE REVIEW

File: outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package io.praesidio.outbox;
+
+import io.praesidio.outbox.values.MessageId;
+
+public interface InternalEventPublisher {
```
Comment (filip-halemba) [inline outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java:5]:
```suggestion
public interface TransactionalEventPublisher {
```
WDYT?

File: outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    void messageReadyForProcessingAfterCommit(MessageId messageId);
```
Comment (filip-halemba) [inline outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java:6]:
```suggestion
    void processAfterCommit(MessageId messageId);
```
WDYT?

File: outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java (hunk 1 above)
Comment (filip-halemba) [inline outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java:5]: Maybe we need to add javadoc?

Comment (mcybulsk) [reply outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java:5]: Doesn't this apply to every public interface we want to be extendable? Or is this interface special in your opinion?

Maybe we just need a new task to do it for each of the interfaces?

Comment (mcybulsk) [reply outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java]: Fine with me :+1:

File: outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java (hunk 2 above)
Comment (mcybulsk) [reply outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java]: Fine with me :+1:

File: outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java (hunk 1 above)
Comment (filip-halemba) [reply outbox-core/src/main/java/io/praesidio/outbox/InternalEventPublisher.java]: Let's do it in another task!

https://github.com/praesidio/outbox/issues/20

**Review by filip-halemba — APPROVED** [review]

**Pull request merged**